src/software/amazon/s3tables/iceberg/S3TablesCatalogOperations.java

```diff
@@ -181,17 +184,32 @@ public void doRefresh() {
      */
     @Override
     public void doCommit(TableMetadata base, TableMetadata metadata) {
-        boolean newTable = false;
+        boolean newTable = base == null;
+        TableMetadata cloneMetadata = metadata;
+        if (newTable && metadata.metadataFileLocation() != null) {
+            String location = createNewTableIfNotExists();
+            /* Because we only write metadata if the location is null we want to trick spark into writing this and returning a new location for us.
+              This is used for stored procedures where they just call DoCommit() to create a table and sometimes pass existing metadata.
+              So we are going to commit a new metadata location for our warehouse to use from the existing schema essentially cloning the metadata for a new table.
+             */
+
+            cloneMetadata = TableMetadata.buildFromEmpty()
+                .addSchema(metadata.schema(), metadata.lastColumnId())
+                .addPartitionSpec(metadata.spec())
+                .setDefaultPartitionSpec(metadata.defaultSpecId())
+                .addSortOrder(metadata.sortOrder())
+                .setDefaultSortOrder(metadata.defaultSortOrderId())
+                .setLocation(location)
+                .build();
```

Comment:
@stubz151, ideally to match the behaviour of the register_table we should be using the same s3 path which is given in the arg which will not be the s3table bucket path

please ref doc  https://iceberg.apache.org/docs/nightly/spark-procedures/#register_table

IIUC the reporter of this issue is writing a catalog migrator from Nessie to S3Catalog, the Nessie catalog will contain reference to s3 general purpose bucket path. We will have to think this through more IMHO !

Reply:
I think s3 tables doesn't support non s3 tables arn's here. Should we just disable the feature instead?